grpc-api/src/main/java/io/deephaven/grpc_api/appmode/ApplicationTicketResolver.java

```diff
@@ -51,66 +51,75 @@ public synchronized void onApplicationLoad(final ApplicationState app) {
+            throw GrpcUtil.statusRuntimeException(Code.FAILED_PRECONDITION,
+                    "Could not resolve '" + logId + "': field '" + getLogNameFor(id)
+                            + "' does not belong to an application");
         }
         final Field<Object> field = id.app.getField(id.fieldName);
         if (field == null) {
-            throw GrpcUtil.statusRuntimeException(Code.NOT_FOUND, "ticket not found: " + getLogNameFor(id));
+            throw GrpcUtil.statusRuntimeException(Code.NOT_FOUND,
+                    "Could not resolve '" + logId + "': field '" + getLogNameFor(id) + "' not found");
         }
         // noinspection unchecked
         return SessionState.wrapAsExport((T) field.value());
     }
 
     @Override
-    public SessionState.ExportObject<Flight.FlightInfo> flightInfoFor(final @Nullable SessionState session,
-            final Flight.FlightDescriptor descriptor) {
-        final AppFieldId id = appFieldIdFor(descriptor);
+    public SessionState.ExportObject<Flight.FlightInfo> flightInfoFor(
+            final @Nullable SessionState session, final Flight.FlightDescriptor descriptor, final String logId) {
+        final AppFieldId id = appFieldIdFor(descriptor, logId);
         if (id.app == null) {
-            throw new IllegalStateException("Cannot resolve field without knowing the application");
+            throw GrpcUtil.statusRuntimeException(Code.FAILED_PRECONDITION,
+                    "Could not resolve '" + logId + "': field does not belong to an application");
         }
 
         final Flight.FlightInfo info;
         synchronized (id.app) {
             Field<?> field = id.app.getField(id.fieldName);
             if (field == null) {
-                throw GrpcUtil.statusRuntimeException(Code.NOT_FOUND, "no flight found: " + getLogNameFor(id));
+                throw GrpcUtil.statusRuntimeException(Code.NOT_FOUND,
+                        "Could not resolve '" + logId + "': field '" + getLogNameFor(id) + "' not found");
             }
             Object value = field.value();
             if (value instanceof Table) {
                 info = TicketRouter.getFlightInfo((Table) value, descriptor, flightTicketForName(id.app, id.fieldName));
             } else {
-                throw GrpcUtil.statusRuntimeException(Code.NOT_FOUND, "flight not found: " + getLogNameFor(id));
+                throw GrpcUtil.statusRuntimeException(Code.NOT_FOUND,
+                        "Could not resolve '" + logId + "': field '" + getLogNameFor(id) + "' is not a flight");
             }
         }
 
         return SessionState.wrapAsExport(info);
     }
 
     @Override
-    public <T> SessionState.ExportBuilder<T> publish(SessionState session, ByteBuffer ticket) {
+    public <T> SessionState.ExportBuilder<T> publish(
+            SessionState session, ByteBuffer ticket, final String logId) {
         throw new UnsupportedOperationException("applications cannot be published to");
```

Comment:
any chance these should get a better message, or is this only reachable via server-side machinery?

Reply:
These are only reachable if you manually create an application ticket, and try to call a method that writes to it. Such as DoPut. If you DoPut to an application field ticket - that's when you hit this.